### PR TITLE
Parallax camera

### PR DESCRIPTION
**Pull request by hukumka**
Move functionality of parallax system into custom projection matrix

Motivation: 
+ It makes sense for parallax effect to be part of projection matrix
+ It does not affect translation of background items, making it easier to add moving background items
+ Marine does not cause background to move then near the edge of the map (then camera is static)

### CODE REVIEW

File: src/entities/camera.rs [hunk 1]
```diff
@@ -1,25 +1,67 @@
...
+
+/// Add parallax translation to camera.
+///
+/// Camera assumed to be setup with `init_parallax_camera`
+/// (Currently equal to `Camera::standard_2d`)
+///
+/// If parallax translation being kept equal to camera translation
+/// it would cause further objects to move away slower (in screen coordinates)
+/// with infinitely distant objects being stationary (infinity=z_far=2000.0)
+pub fn set_paralax_offset(camera: &mut Camera, x: f32, y: f32){
```
Comment (krankur) [inline src/entities/camera.rs]: Nice work! This gets rid of some unnecessary stuff that I was doing. And thanks for adding the comments.

File: src/entities/camera.rs [hunk 2]
```diff
@@ -1,25 +1,67 @@
...
+    let z_near = 0.01;
+    let z_far = 2000.0;
+
+    matrix[(0, 0)] = 2.0/width;
+    matrix[(1, 1)] = -2.0/height;
+    matrix[(2, 2)] = -1.0 / (z_far - z_near);
+    matrix[(2, 3)] = -z_near / (z_far - z_near);
+
+    *camera.as_matrix_mut() = matrix;
+    set_paralax_offset(&mut camera, 0.0, 0.0);
```
Comment (krankur) [inline src/entities/camera.rs]: Nit: change `paralax` to `parallax`

File: src/entities/camera.rs [hunk 3]
```diff
@@ -1,25 +1,67 @@
...
 }
+
+fn init_parallax_camera(width: f32, height: f32) -> Camera{
+    let mut camera = Camera::standard_2d(width, height);
+    // Set orthogonal projection matrix
+    let mut matrix = Matrix4::identity();
+    let z_near = 0.01;
+    let z_far = 2000.0;
+
+    matrix[(0, 0)] = 2.0/width;
```
Comment (krankur) [inline src/entities/camera.rs]: nit: add space around `/` here and on the next line.

**Comment by erlend-sh** [comment]
@hukumka why the close? There are many valid reasons, just wanna know which it is. :)

**Comment by hukumka** [comment]
@erlend-sh I misjudged complexity of solution. It took me three pages to give good
 explanation of how it works ([Parallax_projection_camera.pdf](https://github.com/amethyst/space-menace/files/3538127/Parallax_projection_camera.pdf)), so we came to conclusion that it just not worth to introduce extra complexity. I probably should have commented it then closing

**Comment by krankur** [comment]
@erlend-sh the changes work perfectly and the solution is great. Its even better than the current implementation in that its concise and might even improve the performance of the game (haven't checked it though). However it brings in some mathematical complexity that might be difficult to understand for game dev beginners who don't have a background in mathematics or have been out of touch, like myself. This is why, I and @hukumka decided that we won't merge it, at least for now. In future, if we face any issues with the current implementation, we can reopen and merge this.

**Pull request closed without merging**